### DISCUSSION TITLE
16 Analog input with 4067 MUX

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_basic_synth#111] [Q&A]
Hi marcel license,
In your code you have used 4067 mux IC to control 8 parameter. 4067 Has 16 analogue input.
How can I use 4067 mux to use 16 potentiometers to control 16 midi parameters? What changes I need to do in the code?

**Comment by marcel-licence** [comment]
Hi @souravlal 
you can activate: ADC_TO_MIDI_ENABLED
The code is located here: https://github.com/marcel-licence/esp32_basic_synth/blob/main/adc_module.ino which has internal oversampling for a better performance.
You also need to define ADC_MUL_S0_PIN .. ADC_MUL_S3_PIN depends on the count on channels you want to use.
Hope this helps to get the first step done
